### ISSUE TITLE
DatePicker accessibility with custom Label

### ISSUE THREAD

**Issue by rhublard** [OfficeDev/office-ui-fabric-react#8564]
When using `DatePicker` component with a custom label component based on `office-ui-fabric-react`.

For accessibility reason, when using a given identifier to set DatePicker `id` prop and the label `for` attribute, the component gets rendered with two different ids : `id-example-label` for DatePicker and `id-example` for the label.

We were wondering if there was a workaround possible not implying to format the label `for` to include '-label', as other input components don't have this behavior.

#### Actual behavior:

Label and DatePicker use a different Id/For, provoking accessibility issues.

https://codepen.io/rhublard/pen/bJdVJQ

#### Expected behavior:

Label and DatePicker use a same ID/For.

Thanks.

**Comment by kkjeer** [comment]
@rhublard thanks for reporting this; I've started an internal discussion to determine the best approach here.

**Comment by kkjeer** [comment]
Hi @rhublard, this is intended behavior. The DatePicker has multiple child components, including TextField. We cannot have duplicate ids within the component, but we also want to respect whatever you pass in as the DatePicker's `id` prop. We can't simply use the `id` prop on the TextField as well as the DatePicker's root element, since that would result in two elements in the DOM with the same id. For this reason, we use `id + '-label'` as the id for the TextField.

As you described in the issue, you can use `htmlFor={id + '-label'}` on your Label component.

**Comment by rhublard** [comment]
Hi @kkjeer ,
It makes sense, we will use `htmlFor={id + '-label'}` then. We wanted to make sure before deciding anything on our side.

Thank you for your help.

**Comment by cliffkoh** [comment]
Please feel free to close the issue if you feel your question has been resolved satisfactorily. :)
